ColesStopAndShop/CashRegister.cs

```diff
@@ -218,7 +218,15 @@ public void ProcessPayment(PaymentType paymentChosen)
                 }
                 else
                 {
-                    listOfItems.Add((ItemId)itemScanned);
+                    // This loop checks to verify the key exists in the pre-defined dictionary before adding it.
+                    // No error message is thrown if non existent and nothing is added to list, just moves on to next item.
+                    foreach (int item in storeDeployedAt.AllItemsAtStore.Keys)
+                    {
+                        if (itemScanned == item)
+                        {
+                            listOfItems.Add((ItemId)itemScanned);

```

Comment:
- [ ] Breaking out of the loop once a match is found will speed up execution for when the items sold dictionary grows